contrib/rabbitmq/pkg/apis/sources/v1alpha1/rabbitmq_types.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/knative/pkg/apis"
+	"github.com/knative/pkg/apis/duck"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RabbitmqSource is the Schema for the rabbitmqsources API.
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:categories=all,knative,eventing,sources
+type RabbitmqSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RabbitmqSourceSpec   `json:"spec,omitempty"`
+	Status RabbitmqSourceStatus `json:"status,omitempty"`
+}
+
+var _ runtime.Object = (*RabbitmqSource)(nil)
+
+var _ apis.Immutable = (*RabbitmqSource)(nil)
+
+var _  = duck.VerifyType(&RabbitmqSource{}, &duckv1alpha1.Conditions{})
+
+type RabbitmqSourceExchangeConfigSpec struct {
+	Name        string `json:"name,omitempty"`
+	TypeOf      string `json:"type,omitempty"`
+	Durable     bool   `json:"durable,omitempty"`
+	AutoDeleted bool   `json:"auto_detected,omitempty"`
+	Internal    bool   `json:"internal,omitempty"`
+	NoWait      bool   `json:"nowait,omitempty"`
+}
```

Comment:
It's good to add comments to the properties of RabbitmqSourceExchangeConfigSpec, e.g. description, optional.

Reply:
Hi @daisy-ycguo , thanks for your comment. I have added descriptions & detailed description in [README.md](https://github.com/knative/eventing-sources/blob/f55939e0644c1631ddab8d087fd7b192755d514b/contrib/rabbitmq/samples/README.md#rabbitmq-event-source) point 2.